src/components/FileTree/index.tsx

```diff
@@ -10,14 +10,38 @@ type FileNode = {
   name: string;
 };
 
-type DirectoryNode = {
+export type DirectoryNode = {
   id: string;
   name: string;
   children: TreeNode[];
 };
 
 type TreeNode = FileNode | DirectoryNode;
 
+const recursiveSortInPlace = (node: DirectoryNode): void => {
+  node.children.sort((a, b) => {
+    if ('children' in a && 'children' in b) {
+      return a.name.localeCompare(b.name);
+    }
+
+    if ('children' in a === false && 'children' in b === false) {
+      return a.name.localeCompare(b.name);
```

Comment:
This is the same thing as the result of the other `if` test above. Do you want to `||` them, or do you think it's better this way because it's more readable?

Reply:
I was thinking it was more readable that way